src/simple.jl

```diff
@@ -97,12 +97,12 @@ end
 return: a solution matrix with time steps in `t` along rows and
         state variable `y` along columns
 """
-cvode(f::Function, y0::Vector{Float64}, t::Vector{Float64}, userdata::Any=nothing;
+cvode(f::Function, y0::Vector{Float64}, t::AbstractVector, userdata::Any=nothing;
                 integrator=:BDF, reltol::Float64=1e-3, abstol::Float64=1e-6) = 
     cvode!(f, y0, t, zeros(length(t), length(y0)), userdata; 
                     integrator=integrator, reltol=reltol, abstol=abstol)
 
-function cvode!(f::Function, y0::Vector{Float64}, t::Vector{Float64}, y::Matrix{Float64}, userdata::Any=nothing;
+function cvode!(f::Function, y0::Vector{Float64}, t::AbstractVector, y::Matrix{Float64}, userdata::Any=nothing;
```

Comment:
I'm pretty sure Sundials needs `Vector{Float64}`s since it's doing C interop. Have you tested other `AbstractVector`s?